frontend/src/pages/TrainingReports/index.js

```diff
@@ -171,6 +171,22 @@ export default function TrainingReports({ match }) {
     }
   };
 
+  const onDeleteEvent = async (eventId) => {
+    try {
+      // delete the event
+      await deleteEvent(String(eventId));
+
+      // update the UI, exclude the deleted event.
+      const events = displayEvents.map((e) => ({ ...e })).filter((e) => e.id !== eventId);
+
+      // update the events state.
+      setDisplayEvents(events);
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.log(e);
```

Comment:
Do we want to show a UI error in this case?

Reply:
(or a loading screen)

Reply:
Displayed error.